File: easy/0001_two_sum/solution.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        targets_map = {}

        for i in range(len(nums)):
            other_num = target - nums[i]

            if other_num in targets_map:
                return [targets_map[other_num], i]
            else:
                targets_map[nums[i]] = i

File: easy/0001_two_sum/solution.cpp
#include <unordered_map>
#include <vector>

class Solution {
public:
  std::vector<int> twoSum(std::vector<int> &nums, int target) {
    std::unordered_map<int, int> targetsMap;
    int i, otherNum;

    for (i = 0; i < nums.size(); i++) {
      otherNum = target - nums[i];

      if (targetsMap.find(otherNum) != targetsMap.end())
        break;
      else
        targetsMap[nums[i]] = i;
    }

    return {targetsMap[otherNum], i};
  }
};
